neo4j/neo4j.go

```diff
@@ -0,0 +1,223 @@
+package neo4jstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gofiber/utils/v2"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j/auth"
+	"github.com/neo4j/neo4j-go-driver/v5/neo4j/config"
+)
+
+// Storage interface that is implemented by storage providers
+type Storage struct {
+	db         neo4j.DriverWithContext
+	gcInterval time.Duration
+	done       chan struct{}
+
+	cypherMatch  string
+	cypherMerge  string
+	cypherDelete string
+	cypherReset  string
+	cypherGC     string
+}
+
+type model struct {
+	Key string `json:"k"`
+	Val string `json:"v"`
+	Exp int64  `json:"e"`
+}
+
+type neo4jConnConfig struct {
+	URI            string
+	Auth           auth.TokenManager
+	Configurations []func(*config.Config)
+}
+
+func newDriverWithContext(cfg neo4jConnConfig) (neo4j.DriverWithContext, error) {
+	return neo4j.NewDriverWithContext(cfg.URI, cfg.Auth, cfg.Configurations...)
+}
+
+// New creates a new storage
+func New(config ...Config) *Storage {
+	// Set default config
+	cfg := configDefault(config...)
+
+	// Select db connection
+	var err error
+	db := cfg.DB
+	if db == nil {
+		db, err = newDriverWithContext(neo4jConnConfig{
+			URI:            cfg.URI,
+			Auth:           cfg.Auth,
+			Configurations: cfg.Configurations,
+		})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to create connection pool: %v\n", err)
```

Comment:
This should throw a panic since we can't create connection. That's the behavior in all the drivers.

This can be easily tested using `testify`